LBHackney-IT/lbh-fss-portal-frontend
# Menu navigation permissions

**Description** 

Users with 'viewer' or 'admin' roles should have access to **/users** and **/analytics** pages, and should be able to see **users** and **analytics** menu links

Users without 'viewer' or 'admin' roles (i.e. with 'vcso' role) should not have access to **/users** and **/analytics** pages, and should not be able to see **users** and **analytics** menu links

**Test instructions**

- pull this branch
- navigate to mock-api/mockUsers.json and temporarily delete all users except one, update the user roles with varying combinations of 'viewer',  'admin' and 'vcso' (will need to restart mock api `yarn mock-api` between changes)
- check expected behaviour by viewing menu links and trying to access **/users** and **/analytics**  (see screenshot below of expected view if user does not have permissions to access either **/user** or **/analytics**) 

![image](https://user-images.githubusercontent.com/69238887/90890291-97940c00-e3b1-11ea-9ffa-5a85c7bc4799.png)

- tests have been added for the following components, please run these to make sure they all pass
  - ProtectedLayout
  - AnalyticsDashboard
  - ListUsers
  - UserTable

If running all tests you should see 46 test suites fails and 6 pass (these fails are because of empty test files which have not yet been populated) 

![image](https://user-images.githubusercontent.com/69238887/91063259-ad116c00-e625-11ea-8842-dd0c48bd01dd.png)


**Ticket**

https://app.clickup.com/t/9e5u9e
